src/extension-runners/chromium.js

```diff
@@ -126,8 +174,44 @@ export class ChromiumExtensionRunner {
       chromeFlags.push(...this.params.args);
     }
 
-    if (this.params.chromiumProfile) {
-      chromeFlags.push(`--user-data-dir=${this.params.chromiumProfile}`);
+    // eslint-disable-next-line prefer-const
+    let {userDataDir, profileDirName} =
+      await ChromiumExtensionRunner.getProfilePaths(
+        this.params.chromiumProfile);
+
+    if (userDataDir && this.params.keepProfileChanges) {
+      if (profileDirName
+        && !await ChromiumExtensionRunner.isUserDataDir(userDataDir)) {
+        throw new Error('The profile you provided is not in a ' +
+          'user-data-dir. The changes cannot be kept. Please either ' +
+          'remove --keep-profile-changes or use a profile in a ' +
+          'user-data-dir directory');
+      }
+    } else if (userDataDir) {
+      // the user provided an existing profile directory but doesn't want
+      // the changes to be kept. we copy this directory to a temporary
+      // user data dir.
+      const tmpDir = new TempDir();
+      // TODO: Remove tmpDir - https://github.com/mozilla/web-ext/issues/1957
```

Comment:
Sorry for the confusion, as Luca mentioned at https://github.com/mozilla/web-ext/pull/1920/files#r454511824 , the `tmp` library already removes the file at exit. So this comment can be removed.